File: backend/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid.Models;

public class Question {
    [Key]
    public int Id { get; set; }
    public int Order { get; set; }
    public string Body { get; set; } = "";
    public Quiz Quiz { get; set; } = null!;
    public int QuizId { get; set; }
    public ICollection<Solution> Solutions { get; set; } = new HashSet<Solution>();
    public ICollection<Answer> Answers { get; set; } = new HashSet<Answer>();
    [NotMapped]
    public string QuizName { 
        get {
            return Quiz.Name;
        }
    }
    [NotMapped]
    public string AnswerStatus { get; set; } = "";
    [NotMapped]
    public Answer Answer { get; set; } = null!;
    [NotMapped]
    public int PreviousQuestion { get; set; } = 0;
    [NotMapped]
    public int NextQuestion { get; set; } = 0;
}

public class QuestionDto {
    public int Id { get; set; }
    public int Order { get; set; }
    public string Body { get; set; } = "";
    public int QuizId { get; set; }
    public string QuizName { get; set; } = "";
    public QuizDTO Quiz { get; set; } = null!;
    public string AnswerStatus { get; set; } = "";
    public AnswerDto Answer { get; set; } = null!;
    public int PreviousQuestion { get; set; } = 0;
    public int NextQuestion { get; set; } = 0;
    public ICollection<SolutionDto> Solutions { get; set; } = new HashSet<SolutionDto>();
}

public class QuestionWithSolutionsDto {
    public int Id { get; set; }
    public int Order { get; set; }
    public string Body { get; set; } = "";
    public int QuizId { get; set; }
    public QuizDTO? Quiz { get; set; } = null!;
    public ICollection<SolutionDto> Solutions { get; set; } = new HashSet<SolutionDto>();
}

public class QuestionWithIdDto {
    public int Id { get; set; }
}

File: frontend/src/app/models/question.ts
import { Type } from 'class-transformer';
import { Quiz } from './quiz';
import { Solution } from './solution';
import { Answer } from './answer';

export class Question {
    id?: number;
    order?: number;
    body?: string;
    quizId?: number;
    @Type(() => Quiz)
    quiz?: Quiz;
    answerStatus? : string;
    @Type(() => Answer)
    answer? : Answer;
    previousQuestion?: number;
    nextQuestion?: number;
    sendButtonDisabled?: boolean = true;
    solutions?: Solution[];

    get quizName(): string {
        return this.quiz ? this.quiz.name || 'N/A' : 'N/A';
    }

    get previousButtonDisabled(): boolean {
        return this.previousQuestion == 0;
    }

    get nextButtonDisabled(): boolean {
        return this.nextQuestion == 0;
    }

    get isTest(): boolean {
        return this.quiz?.isTest ?? false;
    }

}